pkg/cfn/manager/nodegroup.go

```diff
@@ -28,6 +34,50 @@ func (c *StackCollection) CreateNodeGroup(seq int, errs chan error) error {
 	return c.CreateStack(name, stack, nil, errs)
 }
 
+func (c *StackCollection) ScaleNodeGroup(errs chan error) error {
+	clusterName := c.makeClusterStackName()
+	c.spec.ClusterStackName = clusterName
+	name := c.makeNodeGroupStackName(0)
+	logger.Info("scaling nodegroup stack %q in cluster %s", name, clusterName)
+
+	if c.spec.MinNodes == 0 && c.spec.MaxNodes == 0 {
+		c.spec.MinNodes = c.spec.Nodes
+		c.spec.MaxNodes = c.spec.Nodes
+	}
+
+	// Get current stack
+	template, err := c.getStackTemplate(name)
+	if err != nil {
+		return errors.Wrapf(err, "error getting stack template %s", name)
+	}
+	logger.Debug("stack template (pre-scale change): %s", template)
+
+	// Update the template
+	//TODO: In the future this needs to use Goformation but at present we manipulate the
+	// JSON directly as the version of goformation we are using isn't handling Refs well
+	var f interface{}
+	err = json.Unmarshal([]byte(template), &f)
+	if err != nil {
+		return errors.Wrap(err, "error pasring JSON")
+	}
+
+	m := f.(map[string]interface{})
+	res := m["Resources"].(map[string]interface{})
+	ng := res["NodeGroup"].(map[string]interface{})
+	props := ng["Properties"].(map[string]interface{})
```

Comment:
I'm not particularly picky, and we can probably assume schema won't change too much, but anyway there's an issue here that if we are to introduce check for each of the keys, it'd be a nightmare...

Reply:
I think this would only be temporary until 'Refs' are handled completely in Goformation (or something else)